ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/ParticipantConfig.scala

```diff
@@ -3,45 +3,37 @@
 
 package com.daml.ledger.runner.common
 
+import com.daml.caching
+import com.daml.ledger.runner.common.ParticipantConfig._
 import com.daml.lf.data.Ref
+import com.daml.platform.apiserver.ApiServerConfig
 import com.daml.platform.configuration.IndexConfiguration
 import com.daml.platform.indexer.IndexerConfig
-import com.daml.ports.Port
-
-import java.nio.file.Path
-import java.time.Duration
+import com.daml.platform.store.LfValueTranslationCache
 
 final case class ParticipantConfig(
-    mode: ParticipantRunMode,
-    participantId: Ref.ParticipantId,
+    participantId: Ref.ParticipantId = DefaultParticipantId,
     // A name of the participant shard in a horizontally scaled participant.
-    shardName: Option[String],
-    address: Option[String],
-    port: Port,
-    portFile: Option[Path],
-    serverJdbcUrl: String,
-    managementServiceTimeout: Duration = ParticipantConfig.DefaultManagementServiceTimeout,
-    indexerConfig: IndexerConfig,
-    apiServerDatabaseConnectionPoolSize: Int =
-      ParticipantConfig.DefaultApiServerDatabaseConnectionPoolSize,
-    apiServerDatabaseConnectionTimeout: Duration =
-      ParticipantConfig.DefaultApiServerDatabaseConnectionTimeout,
-    maxContractStateCacheSize: Long = IndexConfiguration.DefaultMaxContractStateCacheSize,
-    maxContractKeyStateCacheSize: Long = IndexConfiguration.DefaultMaxContractKeyStateCacheSize,
-    maxTransactionsInMemoryFanOutBufferSize: Long =
-      IndexConfiguration.DefaultMaxTransactionsInMemoryFanOutBufferSize,
+    shardName: Option[String] = DefaultShardName,
+    runMode: ParticipantRunMode = DefaultRunMode,
+    indexer: IndexerConfig = DefaultIndexerConfig,
```

Comment:
Would this be better named indexService with the type being changed to `IndexServiceConfig` ?